Documentation/Contributing/UnitTests.md

```diff
@@ -0,0 +1,133 @@
+
+# Writing tests for your code
+To ensure MRTK being a stable and reliable toolkit, every feature should come with unit tests and some application in one of the example scenes.
+
+Preferrably when fixing a bug there should also be a test added to avoid running into the same issue again in the future.
+
+Having good test coverage in a big codebase like MRTK is crucial for stability and having confidence when doing changes in code.
+
+
+MRTK uses the [Unity Test Runner](https://docs.unity3d.com/Manual/testing-editortestsrunner.html) which uses a Unity integration of [NUnit](https://nunit.org/). 
+
+This guide will give a starting point on how to add tests to MRTK. It will not explain the [Unity Test Runner](https://docs.unity3d.com/Manual/testing-editortestsrunner.html) and [NUnit](https://nunit.org/) which can be looked up in the links provided.
+
+There's two types of tests that can be added for new code
+
+* Edit mode tests
+* Play mode tests
+
+## Play mode tests
+
+Play mode tests will be executed in Unity's play mode and should be added into MixedRealityToolkit.Tests > PlaymodeTests. 
+To create a new test the following template can be used: (todo: Setup() and teardown should be moved into a base class of mrtk tests)
+
+``` csharp
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// Unity doesn't include the the required assemblies (i.e. the ones below).
+// Given that the .NET backend is deprecated by Unity at this point it's we have
+// to work around this on our end.
+using NUnit.Framework;
+using UnityEngine.TestTools;
+using System.Collections;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    public class ExampleTest : IPrebuildSetup
+    {
+        public void Setup()
+        {
+            PlayModeTestUtilities.EnsureTextMeshProEssentials();
+        }
+
+
+        [TearDown]
+        public void ShutdownMrtk()
+        {
+            TestUtilities.ShutdownMixedRealityToolkit();
```

Comment:
It's also kinda interesting seeing this shutdown of the MRTK object but no corresponding startup of it - I assume that EnsureTextMeshProEssentials is more about ensuring that package's assets exist, but isn't actually setting up the MRTK object in the scene.

Reply:
i picked a bad example - i'll update this section!

Reply:
added missing comments to textmesh pro and the teardown and added init of mrtk to that template as well